cg/meta/upload/gisaid/gisaid.py

```diff
@@ -102,53 +140,127 @@ def build_gisaid_csv(self, gsaid_samples: List[GisaidSample], file_name: str) ->
-                ),
-                covv_orig_lab=self.lims_api.get_sample_attribute(
-                    lims_id=sample_id, key="original_lab"
-                ),
-                covv_orig_lab_addr=self.lims_api.get_sample_attribute(
-                    lims_id=sample_id, key="original_lab_address"
-                ),
-            )
-            gisaid_samples.append(gisaid_sample)
-        return gisaid_samples
+        log_file = Path(f"{self.gisaid_log_dir}/{case_id}.log")
+        if not log_file.parent.exists():
+            raise ValueError(f"Gisaid log dir: {self.gisaid_log_dir} doesnt exist")
+        return log_file
+
+    def append_log(self, temp_log: Path, gisaid_log: Path) -> None:
+        """appends temp log to gisaid log and delete temp file"""
+
+        with open(str(temp_log.absolute()), "r") as open_temp_log:
+            new_log_data: List = json.load(open_temp_log)
+            if gisaid_log.stat().st_size != 0:
+                with open(str(gisaid_log.absolute()), "r") as open_gisaid_log:
+                    old_log_data: List = json.load(open_gisaid_log)
+                    new_log_data.extend(old_log_data)
 
-    def upload(self, csv_file: Path, fasta_file: Path) -> None:
+        with open(str(gisaid_log.absolute()), "w") as open_gisaid_log:
+            json.dump(new_log_data, open_gisaid_log)
+        temp_log.unlink()
+
+    def upload_results_to_gisaid(self, files: UploadFiles) -> None:
         """Load batch data to GISAID using the gisiad cli."""
 
+        temp_log_file = tempfile.NamedTemporaryFile(dir="/tmp", mode="w+", delete=False)
         load_call: list = [
+            "--logfile",
+            temp_log_file.name,
             "CoV",
             "upload",
             "--csv",
-            str(csv_file.absolute()),
+            str(files.csv_file.absolute()),
             "--fasta",
-            str(fasta_file.absolute()),
+            str(files.fasta_file.absolute()),
         ]
         self.process.run_command(parameters=load_call)
-
+        self.append_log(temp_log=Path(temp_log_file.name), gisaid_log=files.log_file)
+        temp_log_file.close()
         if self.process.stderr:
             LOG.info(f"gisaid stderr:\n{self.process.stderr}")
         if self.process.stdout:
             LOG.info(f"gisaid stdout:\n{self.process.stdout}")
 
     def __str__(self):
         return f"GisaidAPI(dry_run: {self.dry_run})"
+
+    def get_accession_numbers(self, log_file: Path) -> Dict[str, str]:
+        """parse accession numbers and sample ids from log file"""
+
+        LOG.info("Parsing accesion numbers from log file")
+        accession_numbers = {}
+        with open(str(log_file.absolute())) as log_file:
+            log_data = json.load(log_file)
+            for log in log_data:
+                if log.get("code") != "epi_isl_id":
+                    continue
+                accession_obj = GisaidAccession(
+                    accession_nr=log.get("msg"), sample_id=log.get("msg")
```

Comment:
this becomes same value?

Reply:
The same log string from gisaid is being parsed diferently by the validators. I couldnt decide wether to do like this or add another field: log_msg and parse from that field instead... Decided to do,like this. What do you think?

Reply:
This is a bit confusing! I would make a private property for the log and have validators set the sample id and accession nr

Reply:
okej! :)